core/tests/protractor_utils/ProfilePage.js

```diff
@@ -41,14 +41,19 @@ var ProfilePage = function() {
   };
 
   this.expectCurrUserToHaveProfilePhoto = async function() {
-    expect(await currUserProfilePhoto.isPresent()).toBe(true);
+    await waitFor.visibilityOf(
+      currUserProfilePhoto,
+      'Current user profile photo taking too long to display');
   };
 
   this.expectOtherUserToHaveProfilePhoto = async function() {
     expect(await otherUserProfilePhoto.isPresent()).toBe(true);
```

Comment:
Can we use a waitFor here too? Probably `waitFor.presenceOf`

Reply:
Done!